File: WinPaletter/Terminals/WindowsTerminalCopycat.vb
Public Class WindowsTerminalCopycat
    Private Sub WindowsTerminalCopycat_Load(sender As Object, e As EventArgs) Handles Me.Load
        LoadLanguage
        ApplyStyle(Me)
        Icon = WindowsTerminal.Icon

        Try
            ComboBox1.SelectedIndex = 0
        Catch
        End Try

    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If Not String.IsNullOrEmpty(ComboBox1.SelectedItem) Then
            WindowsTerminal.CCat = ComboBox1.SelectedItem.ToString
            DialogResult = DialogResult.OK
        Else
            WindowsTerminal.CCat = Nothing
            DialogResult = DialogResult.Cancel
        End If

        Close()
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        DialogResult = DialogResult.Cancel
        Close()
    End Sub

End Class

File: WinPaletter/Terminals/WindowsTerminalCopycat.cs
﻿using System;
using System.Windows.Forms;

namespace WinPaletter
{
    public partial class WindowsTerminalCopycat
    {
        public WindowsTerminalCopycat()
        {
            InitializeComponent();
        }
        private void WindowsTerminalCopycat_Load(object sender, EventArgs e)
        {
            this.LoadLanguage();
            ApplyStyle(this);
            Icon = Forms.WindowsTerminal.Icon;

            try
            {
                ComboBox1.SelectedIndex = 0;
            }
            catch
            {
            }

        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (!string.IsNullOrEmpty(ComboBox1.SelectedItem.ToString()))
            {
                Forms.WindowsTerminal.CCat = ComboBox1.SelectedItem.ToString();
                DialogResult = DialogResult.OK;
            }
            else
            {
                Forms.WindowsTerminal.CCat = null;
                DialogResult = DialogResult.Cancel;
            }

            Close();
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }

    }
}